src/grpc/stream_infer_handler.cc

```diff
@@ -324,12 +324,14 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
     if (err == nullptr) {
       TRITONSERVER_InferenceTrace* triton_trace = nullptr;
 #ifdef TRITON_ENABLE_TRACING
-      GrpcServerCarrier carrier(state->context_->ctx_.get());
-      auto start_options =
-          trace_manager_->GetTraceStartOptions(carrier, request.model_name());
-      state->trace_ = std::move(trace_manager_->SampleTrace(start_options));
-      if (state->trace_ != nullptr) {
-        triton_trace = state->trace_->trace_;
+      if (trace_manager_) {
```

Comment:
small thing: in some places we use ```<variable> !=nullptr```  - let's choose one for consistency.

Reply:
Made check to be `if(trace_manager != nulltpr)`